### PR TITLE
add logging info

### LINKED ISSUE #68: add structured logs
### Describe the feature

Right now, you have no idea what's going on when you run a command! One should have an idea what is going on when one runs a command

### Describe alternatives you've considered
do nothing

### Who will this benefit?

users!

### Are you interested in contributing this feature?

maybe?

### PR DESCRIPTION
**Pull request by dave-connors-3**
resolves #68 

@nicholasyager tried my hand at using `loguru` for some basic logging messages here. very open to changing how we structure this and whether we use this package at all -- it's pretty straightforward to use, but I have never added logs to a cli app before, so I don't know what best practices to consider.

### CODE REVIEW

File: dbt_meshify/main.py [hunk 1]
```diff
@@ -105,22 +111,32 @@ def add_contract(select, exclude, project_path, selector, public_only=False):
...
         model_catalog = project.get_catalog_entry(model_unique_id)
         meshify_constructor = DbtMeshModelConstructor(
             project_path=project_path, model_node=model_node, model_catalog=model_catalog
         )
-        meshify_constructor.add_model_contract()
+        logger.info(f"Adding contract to model: {model_unique_id}")
+        try:
+            meshify_constructor.add_model_contract()
+            logger.success(f"Successfully added contract to model: {model_unique_id}")
+        except Exception as e:
```
Comment (b-per) [inline dbt_meshify/main.py:137]: I am OK with this for now to add debugging but I have 2 comments:
- what do we want the tool to do when there is an error? stop at the first error or skip the error and continue until the end (which is what we are doing with the `try/except`)
- if possible, we should be specific about the exceptions raised so we can add specific logic/logging: `Exception as e` is the generic one but we could try to be more specific. This is something that can be revisited later.

**Review by b-per — APPROVED** [review]
OK as is but I added 1 comment.

File: dbt_meshify/storage/yaml_editors.py [hunk 2]
```diff
@@ -248,14 +251,18 @@ def add_model_version(
...
+                model_name=self.model_node.name,
+                models_yml=models_yml,
+                prerelease=prerelease,
+                defined_in=defined_in,
+            )
+            # write the updated yml to the file
+            self.file_manager.write_file(yml_path, updated_yml)
+            logger.info("Model version added to model yml")
+        except Exception as e:
+            logger.error(f"Error adding model version to model yml: {e}")
```
Comment (nicholasyager) [inline dbt_meshify/storage/yaml_editors.py:265]: Currently, this try/except eats the exception. Am I correct in believe that we want to re-raise the exception after logging?

**Review by nicholasyager — COMMENTED** [review]
Two main areas for feedback/discussion include:
1. CLI UX for setting debug/log-level.
2. Exceptions being swalloed by the try/except behavior.

**Comment by dave-connors-3** [comment]
@nicholasyager @b-per made a couple updates yesterday that I think address the question of try/except -- in the except branch, i now include `logger.exception(e)`, which I still raises the exception, as Nich mentioned in their comment. Let me know if that is not sufficient! 

@b-per -- re: more detailed excpetions, i totally agree, but wasn't positive which class was more representative of what the error might be at the level I placed the logs.

**Comment by dave-connors-3** [comment]
@nicholasyager @b-per i am going to merge this as is so I can add logging to PR #63 before I leave to go OOO. I opened #92 to reflect the outstanding discussion points from above!

**Pull request merged**